### ISSUE TITLE
Adding strain to taxonomic rank in kaijuReport

### ISSUE THREAD

**Issue by mytluo** [bioinformatics-centre/kaiju#84]
Hi,

Would it be possible to have the rank go down to strain level when generating the summary report (i.e., `-r strain`)? I see `addTaxonNames` will give the strain when possible; would it be possible to merge/include it in the summary report?

**Comment by pmenzel** [comment]
The NCBI Taxonomy tree only goes down to species level, see their nodes.dmp file.
I don't know how to get reliable strain annotations for the reference genomes..

**Comment by mytluo** [comment]
After adding the line (I commented out the original line, just in case) and re-running `make`, I get this error:
```
$ ~/kaiju/src/kaijuReport -t ~/kaiju/kaijudb/nodes.dmp -n ~/kaiju/kaijudb/names.dmp -i S1_fullout -m 0.001 -o S1_full.summary -v -p
Error: Please specify the rank (phylum, class, order, family, genus, or species) with the -r option.

Kaiju 1.6.2
Copyright 2015-2017 Peter Menzel, Anders Krogh
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

Usage:
   /home/ubuntu/kaiju/src/kaijuReport -t nodes.dmp -n names.dmp -i kaiju.out -o kaiju.report

Mandatory arguments:
   -i FILENAME   Name of input file
   -o FILENAME   Name of output file.
   -t FILENAME   Name of nodes.dmp file
   -n FILENAME   Name of names.dmp file.
   -r STRING     Taxonomic rank, must be one of: phylum, class, order, family, genus, species

Optional arguments:
   -m FLOAT      Number in [0, 100], denoting the minimum required percentage for the taxon to be reported (default: 0.0)
   -c INT        Integer number > 0, denoting the minimum required number of reads for the taxon to be reported (default: 0)
   -u            Unclassified reads are not counted for the total reads when calculating percentages for classified reads.
   -p            Print full taxon path.
   -l            Print taxon path containing only ranks specified by a comma-separated list,
                 for example: superkingdom,phylum,class,order,family,genus,species
   -v            Enable verbose output.

Only one of the options -m and -c may be used at a time.
```
removing the `-m` arg still returned the same error :(

**Comment by mytluo** [comment]
even the original `kaijuReport.cpp` source file (redownloaded) gives me the same error... I can't run `kaijuReport` anymore :(

**Comment by pmenzel** [comment]
Well, the error message says `Error: Please specify the rank (phylum, class, order, family, genus, or species) with the -r option.`. Did you try to add `-r species` to your command?

**Comment by mytluo** [comment]
Yep, I tried it with `-r` and other combinations, as well as no flags at all, and I get the exact same error every time.

**Comment by mytluo** [comment]
I will probably end up reinstalling `kaiju`, but unfortunately there still doesn't seem to be a way to merge the output from `kaijuReport` and `addTaxonNames` :(

**Comment by mytluo** [comment]
A combination of `-p` and `-r` worked! I apologize for not trying that before; last time I tried it it gave me an error saying I could only use one or the other, but that may have been for a different command.

**Comment by pmenzel** [comment]
Great that it works now for you. :)

**Comment by mytluo** [comment]
thank you @pmenzel!

**Comment by pmenzel** [comment]
Hi.

> but it didn't quite work out the way I'd hope  

that's not very specific :)
If you want me to have a look at the code, then please commit your kaiju fork to GitHub. 
Also make a minimal test data set with nodes.dmp, the kaiju database and a test input file together with the expected output.

**Comment by mytluo** [comment]
Hi @pmenzel, apologies for the late reply! I'm working on an internship project that is quickly coming to an end so I've had to redirect my focus; I also lost the AWS instance that had those changes ^^;; 

If I recall correctly though, what I had done was loop through the taxIDs in the `nodes.dmp` file that were given "no rank", search with `eutils` for the parent taxID (if it was also "no rank" it'll search for that ID's parent taxID, etc), and if the parent was "species" or "subspecies" then the "no rank" was replaced with a custom designation. I also modified in `kaijuReport` to include in `rank.compare` that custom designation. I think that was the majority of the modifications I made... I don't remember if the output changed drastically in some way as well due to the loss of the instance I was working in, but I'd imagine I'm missing several steps before that would work properly.
